apps/admin/frontend/src/components/write_in_adjudication_button.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { CheckboxButton, Icons, SearchSelect } from '@votingworks/ui';
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+
+  /* Prevent the CheckboxButton bottom border from being cutoff on hover
+   * from the SearchSelect, which has a higher z-index */
+  button:focus-visible {
+    z-index: 10;
+  }
+`;
+
+export function WriteInAdjudicationButton({
+  caption,
+  isFocused,
+  isSelected,
+  hasInvalidEntry,
+  onChange,
+  onInputFocus,
+  onInputBlur,
+  toggleVote,
+  value,
+  officialCandidateNames,
+  writeInCandidateNames,
+}: {
+  caption?: React.ReactNode;
+  isFocused: boolean;
+  isSelected: boolean;
+  hasInvalidEntry: boolean;
+  // newVal can be name of new or existing candidate,
+  // keyword 'invalid' for invalid write-in, or undefined
+  onChange: (newVal?: string) => void;
+  onInputBlur: () => void;
+  onInputFocus: () => void;
+  toggleVote: () => void;
+  value: string;
+  officialCandidateNames: string[];
+  writeInCandidateNames: string[];
+}): JSX.Element {
+  const theme = useTheme();
+  const [inputValue, setInputValue] = useState('');
+  const inputValueLowerCase = inputValue.toLowerCase();
+
+  function onKeyPress(val?: string) {
+    return setInputValue(val || '');
+  }
+
+  const filteredCandidateOptions = inputValue
+    ? officialCandidateNames.filter((name) =>
+        name.toLowerCase().includes(inputValueLowerCase)
+      )
+    : officialCandidateNames;
+  const filteredWriteInCandidateOptions = inputValue
+    ? writeInCandidateNames.filter((name) =>
+        name.toLowerCase().includes(inputValueLowerCase)
+      )
+    : writeInCandidateNames;
+  const options = filteredWriteInCandidateOptions
+    .concat(filteredCandidateOptions)
+    .map((name) => ({
+      label: name,
+      value: name,
+    }));
+
+  // 'Add: NEW_CANDIDATE' entry if there is no exact match
+  if (
+    inputValue &&
+    !options.find(
+      (item) => item.label.toLowerCase() === inputValue.toLowerCase()
+    )
+  ) {
+    options.push({ label: `Add: ${inputValue}`, value: inputValue });
+  }
+
+  // If value has been entered and it is a new entry, add it the dropdown
+  if (value && !options.find((option) => option.label === value)) {
```

Comment:
Use `Array::some` to make it clear we're checking for a value matching a predicate, not finding a value and using its truthiness.

```suggestion
  if (value && !options.some((option) => option.label === value)) {
```